x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java

```diff
@@ -267,6 +294,11 @@ public LifecyclePolicyMetadata read(StreamInput in, String key) {
 
     @Nullable
     public Step getStep(final IndexMetadata indexMetadata, final Step.StepKey stepKey) {
+        final Tuple<IndexMetadata, Step> cachedStep = cachedSteps.get(indexMetadata.getIndex());
+        if (cachedStep != null && cachedStep.v1() == indexMetadata && cachedStep.v2().getKey().equals(stepKey)) {
```

Comment:
Can you add a comment about why instance equality is used here and okay rather than object equality?

Reply:
👍, 68161e8a3b8168c5d9a64a55fe7bbbd30f0c8afe